documents/serializers/document.py

```diff
@@ -50,7 +59,29 @@ def update(self, document, validated_data):
         if document.locked_after and document.locked_after <= now():
             raise DocumentLockedException()
 
-        return super().update(document, validated_data)
+        user_id = self.initial_data.get("user_id", None)
+        if user_id:
+            if document.user_id:
+                raise PermissionDenied(
+                    detail="Document owner can not be changed.",
+                    code="invalid field: user_id",
+                )
+
+            user, created = User.objects.get_or_create(
+                uuid=user_id, defaults={"username": f"User-{user_id}"}
+            )
+            document.user = user
+            document.save()
+
+        return super().update(document, validated_data) if validated_data else document
```

Comment:
Why there is this validated_data if validated_data else document?

Reply:
If patch data is empty the object is saved regardless which changes updated_at timestamp even when nothing is actually updated. Not sure if it actually matters, but I thought it would be good to not change the timestamp if nothing has actually changed. In real use cases empty patch data might not actually happen at all, but I found this issue while testing.